src/main/java/net/dv8tion/jda/core/handle/UserUpdateHandler.java

```diff
@@ -43,54 +43,70 @@ protected Long handleInternally(JSONObject content)
+        String email = !content.isNull("email") ? content.getString("email") : null; // TODO: client fields
+        Boolean mobile = content.has("mobile") ? content.getBoolean("mobile") : null; // mobile device 
+        Boolean premium = content.has("premium") ? content.getBoolean("premium") : null; // nitro
+        String phoneNumber = !content.isNull("phone") ? content.getString("phone") : null; // verified phone number (verification level !)
 
         if (!Objects.equals(name, self.getName()) || !Objects.equals(discriminator, self.getDiscriminator()))
         {
             String oldName = self.getName();
             String oldDiscriminator = self.getDiscriminator();
             self.setName(name);
             self.setDiscriminator(discriminator);
-            api.getEventManager().handle(
-                    new SelfUpdateNameEvent(
-                            api, responseNumber,
-                            oldName, oldDiscriminator));
+            api.getEventManager().handle(new SelfUpdateNameEvent(api, responseNumber, oldName, oldDiscriminator));
         }
+
         if (!Objects.equals(avatarId, self.getAvatarId()))
         {
             String oldAvatarId = self.getAvatarId();
             self.setAvatarId(avatarId);
-            api.getEventManager().handle(
-                    new SelfUpdateAvatarEvent(
-                            api, responseNumber,
-                            oldAvatarId));
+            api.getEventManager().handle(new SelfUpdateAvatarEvent(api, responseNumber, oldAvatarId));
         }
+
         if (verified != null && verified != self.isVerified())
         {
             boolean wasVerified = self.isVerified();
             self.setVerified(verified);
-            api.getEventManager().handle(
-                    new SelfUpdateVerifiedEvent(
-                            api, responseNumber,
-                            wasVerified));
+            api.getEventManager().handle(new SelfUpdateVerifiedEvent(api, responseNumber, wasVerified));
         }
+
         if (mfaEnabled != null && mfaEnabled != self.isMfaEnabled())
         {
             boolean wasMfaEnabled = self.isMfaEnabled();
             self.setMfaEnabled(mfaEnabled);
-            api.getEventManager().handle(
-                    new SelfUpdateMFAEvent(
-                            api, responseNumber,
-                            wasMfaEnabled));
+            api.getEventManager().handle(new SelfUpdateMFAEvent(api, responseNumber, wasMfaEnabled));
         }
+
         if (api.getAccountType() == AccountType.CLIENT && !Objects.equals(email, self.getEmail()))
         {
             String oldEmail = self.getEmail();
             self.setEmail(email);
-            api.getEventManager().handle(
-                    new SelfUpdateEmailEvent(
-                            api, responseNumber,
-                            oldEmail));
+            api.getEventManager().handle(new SelfUpdateEmailEvent(api, responseNumber, oldEmail));
```

Comment:
Why did you change these events here? They have a consistent setup...

Reply:
^ I wanna reiterate this comment, please don't change these as we have a consistent style for these event handling calls.